### PR TITLE
LEP-259 - Allow employment attribute "prisoner_levy"

### PR DESCRIPTION
**Pull request by ripan**
https://dsdmoj.atlassian.net/browse/LEP-259

---

## Checklists

Author: (before you ask people to review this PR)

- [x] Diff - review it, ensuring it contains only expected changes
- [x] Whitespace changes - avoid if possible, because they make diffs harder to read and conflicts more likely
- [x] Changelog - add a line, if it meets the criteria
- [x] Secrets - no secrets should be added
- [x] Commit messages - say *why* the change was made
- [x] PR description - summarizes *what* changed and *why*, with a JIRA ticket ID
- [ ] Tests pass - on CircleCI
- [x] Conflicts - resolve if Github reports them. e.g. with `git rebase main`

Reviewers remember:

- [ ] Jira ticket criteria are met
- [ ] Migrations - test migration and rollback: `rake db:migrate && rake db:rollback`

### CODE REVIEW

File: app/lib/swagger_docs.rb [hunk 1]
```diff
@@ -1716,6 +1747,11 @@ def schema
                     description: "Amount of national insurance paid - normally negative, but can be positive for a refund",
                     example: -5.24,
                   },
+                  prisoner_levy: {
```
Comment (starswan) [inline app/lib/swagger_docs.rb:1750]: an unfortunate duplication - is it possible to remove?

File: lib/integration_helpers/test_case/employment_collection.rb [hunk 2]
```diff
@@ -84,6 +84,8 @@ def transform_row_to_hash(row, payment_hash)
         payment_hash[:tax] = row[3]
       when "national insurance"
         payment_hash[:national_insurance] = row[3]
+      when "prisoner levy"
+        payment_hash[:prisoner_levy] = row[3]
```
Comment (starswan) [inline lib/integration_helpers/test_case/employment_collection.rb:88]: don't think we need this change - we're not adding prisoner_levy to our spreadsheet tests?

File: lib/integration_helpers/test_case/v5/result_comparer.rb [hunk 3]
```diff
@@ -30,6 +30,7 @@ def call
         employment_income_benefits_in_kind: ->(o) { { actual: o.actual_employment_income[:benefits_in_kind], expected: o.expected_employment_income_benefits_in_kind } },
         employment_income_tax: ->(o) { { actual: o.actual_employment_income[:tax], expected: o.expected_employment_income_tax } },
         employment_income_nic: ->(o) { { actual: o.actual_employment_income[:national_insurance], expected: o.expected_employment_income_nic } },
+        # TODO: : employment_income_prisoner_levy: ->(o) { { actual: o.actual_employment_income[:prisoner_levy], expected: o.expected_employment_income_prisoner_levy } },
```
Comment (starswan) [inline lib/integration_helpers/test_case/v5/result_comparer.rb:33]: see above - no prisoner levy in spreadsheet tests

File: app/lib/swagger_docs.rb [hunk 4]
```diff
@@ -1487,6 +1513,11 @@ def schema
...
+                    oneOf: [{ "$ref" => SCHEMA_COMPONENTS[:currency] }], # "oneOf" hack
+                    description: "Amount of prisoner levy paid - normally negative, but can be positive for a refund",
```
[the remark this answers is not in the document]
Comment (ripan) [reply app/lib/swagger_docs.rb:1518]: Description Updated

File: app/lib/swagger_docs.rb (hunk 1 above)
Comment (ripan) [reply app/lib/swagger_docs.rb:1750]: Actually not. Income object in `employment_details.income` has an additional required attribute `receiving_only_statutory_sick_or_maternity_pay`

**Review by starswan — APPROVED** [review]

**Pull request merged**